src/hooks/useCart/useCart.tsx

```diff
@@ -131,148 +111,48 @@ const mergeWithCart = (
-    );
-  }
-
-  return filteredQuota;
-};
-
-const hasNoQuota = (quota: Quota): boolean =>
-  quota.remainingQuota.every(item => item.quantity === 0);
-
-const hasInvalidQuota = (quota: Quota): boolean =>
-  // Note: Invalid quota refers to negative quota received
-  quota.remainingQuota.some(item => item.quantity < 0);
-
 export const useCart = (
   ids: string[],
   authKey: string,
-  endpoint: string
+  endpoint: string,
+  cartQuota?: ItemQuota[]
 ): CartHook => {
-  const prevIds = usePrevious(ids);
-  const { products, getProduct } = useContext(ProductContext);
-  const prevProducts = usePrevious(products);
   const [cart, setCart] = useState<Cart>([]);
   const [cartState, setCartState] = useState<CartState>("DEFAULT");
   const [checkoutResult, setCheckoutResult] = useState<PostTransactionResult>();
   const [error, setError] = useState<Error>();
-  const [quotaResponse, setQuotaResponse] = useState<Quota | null>(null);
-  const [allQuotaResponse, setAllQuotaResponse] = useState<Quota | null>(null);
   const clearError = useCallback((): void => setError(undefined), []);
+  const { products, getProduct } = useContext(ProductContext);
+  const prevProducts = usePrevious(products);
+  const prevIds = usePrevious(ids);
+  const prevCartQuota = usePrevious(cartQuota);
 
   /**
-   * Fetch quota whenever IDs change.
-   */
-  useEffect(() => {
-    const fetchQuota = async (): Promise<void> => {
-      setCartState("FETCHING_QUOTA");
-      try {
-        const allQuotaResponse = await getQuota(ids, authKey, endpoint);
-        setAllQuotaResponse(allQuotaResponse);
-        const quotaResponse = filterQuotaWithAvailableProducts(
-          allQuotaResponse,
-          products
-        );
-        if (hasInvalidQuota(quotaResponse)) {
-          Sentry.captureException(
-            `Negative Quota Received: ${JSON.stringify(
-              quotaResponse.remainingQuota
-            )}`
-          );
-          setCartState("NO_QUOTA");
-        } else if (hasNoQuota(quotaResponse)) {
-          setCartState("NO_QUOTA");
-        } else {
-          setCartState("DEFAULT");
-        }
-        setQuotaResponse(quotaResponse);
-      } catch (e) {
-        if (e instanceof NotEligibleError) {
-          setCartState("NOT_ELIGIBLE");
-          return;
-        } else {
-          setError(e);
-        }
-        setCartState("DEFAULT");
-      }
-    };
-
-    if (prevIds !== ids || prevProducts !== products) {
-      fetchQuota();
-    }
-  }, [authKey, endpoint, ids, prevIds, prevProducts, products]);
-
-  /**
-   * Merge quota response with current cart whenever quota response or products change.
+   * Update the cart when:
+   *  1. First quota is retrieved on initialisation
+   *  2. When quota is updated (i.e. products or ids change)
    */
   useEffect(() => {
-    if (quotaResponse) {
-      // Note that we must use a callback within this setState to avoid
-      // having cart as a dependency which causes an infinite loop.
```

Comment:
I think we should leave this in for future devs to take note of the potential infinite loop

Reply:
I agree, this has been returned to the respective location!